socorro/external/crashstorage_base.py

```diff
@@ -467,6 +467,47 @@ def __str__(self):
         return ','.join(output)
 
 
+class StorageNamespaceList(collections.Sequence):
+    """A sequence of configuration namespaces for crash stores.
+
+    This acts mostly as a list of strings that correspond to
+    destination.* config keys containing crash store configs. But it
+    also includes a required_config that tells configman how to find
+    the class storage classes, allowing them to define their own
+    required_configs.
+    """
```

Comment:
I'm not really happy with how the magic is explained here. The gist is that we need the value to have a `required_config` that includes the crash storage classes so configman resolves them and sees their `required_config`s during the option overlay stuff.

Reply:
Also, I couldn't figure out a clean way to not need this separate config value with names. It'd be nice to not have to specify it and instead just let Socorro detect `destination.storages.postgres` as a crash storage or something.

Reply:
Pretty sure we need to specify the order of crash storage destinations. That's easily done with a list. There are probably other ways we can specify order, too.

I say "pretty sure" because the way PolyCrashStorage works isn't transactional and we need the crash data to exist in certain places before it exists in others. For example, it needs to exist in S3 before Elasticsearch. It's entirely possible that the code in PolyCrashStorage and the crash storage classes themselves is such that Socorro doesn't support this requirement. I don't think I've looked closely at that, yet.

Reply:
I believe PolyCrashStorage supports writing to storages in order, but does not stop writing if a storage fails during the write, so we could still end up in a situation where ES has the crash but S3 doesn't.